config/runtime/config.go

```diff
@@ -0,0 +1,42 @@
+package runtime
+
+import "github.com/avenga/couper/config"
+
+type Config struct {
+	File      string `env:"config_file"`
+	LogFormat string `env:"log_format"`
+}
+
+// NewConfig creates the runtime config which could be overridden in order:
+// internal.defaults -> config.settings -> flag.args -> env.vars
+func NewConfig(c *config.Gateway) *Config {
+	defaultConf := *DefaultConfig
+	conf := &defaultConf
+	if c != nil && c.Settings != nil {
+		conf.Merge(newConfigFrom(c.Settings))
+	}
+
+	return conf
+}
+
+func newConfigFrom(s *config.Settings) *Config {
+	return &Config{
+		LogFormat: s.LogFormat,
+	}
+}
+
+func (c *Config) Merge(o *Config) *Config {
+	if o == nil {
```

Comment:
`if o == nil` => `if o == nil || c == nil`?

Reply:
`c` ist der _callee_ der methode. kann der jemals `nil` sein?

Reply:
ja:

```
var c *Config
c.Merge(...)
```

Reply:
`c` could be `nil` but should not since `Merge` needs some kind of a base configuration (`DefaultConfig...`) a `panic` if `c` is `nil` would be the right thing here. Since this `c` related panic would be thrown anyways no change is required.